install-extensions.sh

```diff
@@ -4,51 +4,100 @@
 function download_extension() {
   URL=$1
   EXTENSION=$2
-  DOWNLOAD_FILE="${ADDITIONAL_LIBS_DIR}geoserver-${GEOSERVER_VERSION}-${EXTENSION}-plugin.zip"
+  VERSION=$3
+  DOWNLOAD_FILE="${ADDITIONAL_LIBS_DIR}geoserver-${VERSION}-${EXTENSION}-plugin.zip"
 
   if [ -e "$DOWNLOAD_FILE" ]; then
       echo "$DOWNLOAD_FILE already exists. Skipping download."
   else
-    if curl --output /dev/null --silent --head --fail "${URL}"; then
-        echo -e "\nDownloading ${EXTENSION} extension from ${URL} to ${DOWNLOAD_FILE}"
-        wget --progress=bar:force:noscroll -c --no-check-certificate "${URL}" -O ${DOWNLOAD_FILE}
-      else
-        echo "URL does not exist: ${URL}"
+    echo -e "\nDownloading ${EXTENSION} extension from ${URL} to ${DOWNLOAD_FILE}"
+    wget --progress=bar:force:noscroll -c $WGET_OPTS "${URL}" -O ${DOWNLOAD_FILE}
+    if [ "$?" != 0 ]; then
+        echo "ERROR downloading: ${URL}"
     fi
   fi
 }
 
-# Download stable plugins only if INSTALL_EXTENSIONS is true
-if [ "$INSTALL_EXTENSIONS" = "true" ]; then
+function install_lib() {
+    ADDITIONAL_LIB=$1
+    if [ ! -e "$ADDITIONAL_LIB" ]; then
+      echo "Skipping ${ADDITIONAL_LIB}: file not found."
+      return
+    fi
+
+    if [[ $ADDITIONAL_LIB == *.zip ]]; then
+      unzip -q -o -d ${GEOSERVER_LIB_DIR} ${ADDITIONAL_LIB} "*.jar"
+      echo "Installed all jar files from ${ADDITIONAL_LIB}"
+    elif [[ $ADDITIONAL_LIB == *.jar ]]; then
+      cp ${ADDITIONAL_LIB} ${GEOSERVER_LIB_DIR}
+      echo "Installed ${ADDITIONAL_LIB}"
+    else
+      echo "Skipping ${ADDITIONAL_LIB}: unknown file extension."
+    fi
+}
+
+# Download plugins only if DOWNLOAD_EXTENSIONS is true
+if [ "$DOWNLOAD_EXTENSIONS" = "true" ]; then
   echo "Starting download of extensions"
   if [ ! -d "$ADDITIONAL_LIBS_DIR" ]; then
     mkdir -p $ADDITIONAL_LIBS_DIR
   fi
   for EXTENSION in $(echo "${STABLE_EXTENSIONS}" | tr ',' ' '); do
     URL="${STABLE_PLUGIN_URL}/geoserver-${GEOSERVER_VERSION}-${EXTENSION}-plugin.zip"
-    download_extension ${URL} ${EXTENSION}
-  done
-  for EXTENSION in $(echo "${COMMUNITY_EXTENSIONS}" | tr ',' ' '); do
-    URL="${COMMUNITY_PLUGIN_URL}/geoserver-${GEOSERVER_VERSION}-${EXTENSION}-plugin.zip"
-    download_extension ${URL} ${EXTENSION}
+    download_extension ${URL} ${EXTENSION} ${GEOSERVER_VERSION}
   done
+  if [ ${#COMMUNITY_EXTENSIONS} -gt 0 ]; then
+    # build community version string from GEOSERVER_VERSION by removing the last part and adding SNAPSHOT
+    if [ -z "$COMMUNITY_EXTENSIONS_VERSION" ]; then
+      COMMUNITY_EXTENSIONS_VERSION="${GEOSERVER_VERSION/-SNAPSHOT/.x}"
+      COMMUNITY_EXTENSIONS_VERSION="${COMMUNITY_EXTENSIONS_VERSION%.*}-SNAPSHOT"
+    fi
+    COMMUNITY_PLUGIN_BASE_URL=${COMMUNITY_PLUGIN_BASE_URL:-"https://build.geoserver.org/geoserver"}
```

Comment:
Unlike others (like `COMMUNITY_PLUGIN_URL`) this new env `COMMUNITY_PLUGIN_BASE_URL` is not declared in the Dockerfile. I think it makes sense to also add it there for the sake of completeness (and to avoid confusion)